app/javascript/src/lib/codeActions.js

```diff
@@ -0,0 +1,96 @@
+import { Decoration, ViewPlugin, ViewUpdate, WidgetType } from "@codemirror/view"
+
+/**
+ * @typedef CodeAction
+ * @property {number} position
+ * @property {string} label
+ * @property {Function} run
+ */
+
+/**
+ * @typedef {(update: ViewUpdate) => CodeAction[]} CodeActionProvider
+ */
+
+class CodeActionLightBulbWidget extends WidgetType {
+  constructor(actions) {
+    super()
+
+    this.actions = actions
+  }
+
+  toDOM() {
+    const container = document.createElement("span")
+    container.className = "button cm-codeActionsLightbulb"
```

Comment:
Since we can pick the classname here we might as well pick something nicer, even if it doesn't quite follow codemirror conventions

Reply:
I saw we already had some custom stuff using the `cm-` notation and just went with it as well.
But what would you suggest?

Reply:
I think it makes sense to use classnames of our own here and keep `cm-` to codemirror specific things so we can always differentiate between the two

Reply:
Would it make sense to keep that in codeMirror.scss, or should I create a separate file for the extension?

Reply:
Hmm good question, do what you think is best. That file is a bit messy unfortunately

Reply:
For now I'll just leave it in `_codemirror.scss`. Splitting that file up can be done in another PR :)

I also went with `.code-action-lightbulb` for the class name.